### PR TITLE
Update API specs 2020-04-01

### PR DESCRIPTION
**Pull request by digavino**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/frontdoor/resource-manager/Microsoft.Network/stable/2020-04-01/examples/FrontdoorCreate.json [hunk 1]
```diff
@@ -0,0 +1,568 @@
+{
+  "parameters": {
+    "api-version": "2020-01-01",
```
Comment (MoustafaEid) [inline specification/frontdoor/resource-manager/Microsoft.Network/stable/2020-04-01/examples/FrontdoorCreate.json:3]: `2020-04-01`

File: specification/frontdoor/resource-manager/Microsoft.Network/stable/2020-04-01/examples/FrontdoorGet.json [hunk 2]
```diff
@@ -0,0 +1,214 @@
+{
+  "parameters": {
+    "api-version": "2020-01-01",
```
Comment (MoustafaEid) [inline specification/frontdoor/resource-manager/Microsoft.Network/stable/2020-04-01/examples/FrontdoorGet.json:3]: Same as above

File: specification/frontdoor/resource-manager/Microsoft.Network/stable/2020-04-01/frontdoor.json [hunk 3]
```diff
@@ -0,0 +1,2350 @@
...
+          "enum": [
+            "Pending",
+            "Approved",
+            "Rejected",
+            "Disconnected",
+            "Timeout"
+          ],
+          "x-ms-enum": {
+            "name": "PrivateEndpointStatus",
+            "modelAsString": false
```
Comment (MoustafaEid) [inline specification/frontdoor/resource-manager/Microsoft.Network/stable/2020-04-01/frontdoor.json:1267]: This is a bug from the previous version, please set `modelAsString` to `true`

File: specification/frontdoor/resource-manager/Microsoft.Network/stable/2020-04-01/webapplicationfirewall.json [hunk 4]
```diff
@@ -0,0 +1,902 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2019-10-01",
```
Comment (MoustafaEid) [inline specification/frontdoor/resource-manager/Microsoft.Network/stable/2020-04-01/webapplicationfirewall.json:4]: `2020-04-01`

File: specification/frontdoor/resource-manager/readme.md [hunk 5]
```diff
@@ -49,6 +49,24 @@ directive:
...
+### Tag: package-2020-04
+
+These settings apply only when `--tag=package-2020-04` is specified on the command line.
+
+``` yaml $(tag) == 'package-2020-04'
+input-file:
+- Microsoft.Network/stable/2020-01-01/network.json
+- Microsoft.Network/stable/2019-11-01/networkexperiment.json
+- Microsoft.Network/stable/2020-04-01/frontdoor.json
+- Microsoft.Network/stable/2020-04-01/webapplicationfirewall.json
```
Comment (MoustafaEid) [inline specification/frontdoor/resource-manager/readme.md:61]: Please update `package-2020-05` above with the new version

**Review by MoustafaEid — CHANGES_REQUESTED** [review]

File: specification/frontdoor/resource-manager/readme.md (hunk 5 above)
Comment (digavino) [reply specification/frontdoor/resource-manager/readme.md:61]: I think I already did... Did I miss something else ?

Comment (MoustafaEid) [reply specification/frontdoor/resource-manager/readme.md:61]: This line in the same readme file: https://github.com/Azure/azure-rest-api-specs/pull/9383/files/00f8c8a3ea5dbaffcb1bccaa768f77024628d3d7#diff-02bd54d5afdfb095a174499ad32a314bR43
```

``` yaml $(tag) == 'package-2020-05'
input-file:
- Microsoft.Network/stable/2020-05-01/network.json
- Microsoft.Network/stable/2019-11-01/networkexperiment.json
- Microsoft.Network/stable/2020-05-01/frontdoor.json
- Microsoft.Network/stable/2019-10-01/webapplicationfirewall.json
```

**Review by MoustafaEid — APPROVED** [review]

**Comment by MoustafaEid** [comment]
@qianwens we're not sure why the `automation - sdk (SDK Go)` check keeps failing, can you take a look?

File: specification/frontdoor/resource-manager/readme.go.md [hunk 6]
```diff
@@ -20,7 +20,18 @@ batch:
...
+### Tag: package-2020-04 and go
+
+These settings apply only when `--tag=package-2020-04 --go` is specified on the command line.
+Please also specify `--go-sdk-folder=<path to the root directory of your azure-sdk-for-go clone>`.
+
+``` yaml $(tag) == 'package-2020-04' && $(go)
+output-folder: $(go-sdk-folder)/services/$(namespace)/mgmt/2020-04-01/$(namespace)
+```
+
 ```
```
Comment (qianwens) [inline specification/frontdoor/resource-manager/readme.go.md:35]: remove ```?

File: specification/frontdoor/resource-manager/Microsoft.Network/stable/2020-04-01/webapplicationfirewall.json [hunk 7]
```diff
@@ -0,0 +1,902 @@
...
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/FrontDoorWebApplicationFirewallPolicies": {
```
Comment (qianwens) [inline specification/frontdoor/resource-manager/Microsoft.Network/stable/2020-04-01/webapplicationfirewall.json:37]: >FrontDoorWebApplicationFirewallPolicies [](start = 100, length = 39)

frontDoorWebApplicationFirewallPolicies, first character should be lower case

**Review by qianwens — APPROVED** [review]

**Pull request merged**